v0/destinations/mautic/transform.js

```diff
@@ -0,0 +1,244 @@
+  }
+};
+
+/**
+ *
+ * @param {*} message
+ * @param {*} Config
+ * @param {*} endPoint
+ * @returns builded response from responseBuilder
+ */
+const groupResponseBuilder = async (message, Config, endPoint) => {
+  let groupClass;
+  validateGroupCall(message);
+  switch (message.traits.type.toLowerCase()) {
+    case "segments":
+      groupClass = "segments";
+      break;
+    case "campaigns":
+      groupClass = "campaigns";
+      break;
+    case "companies":
+      groupClass = "companies";
+      break;
+    default:
+      throw new CustomError(
+        "This grouping is not supported. Supported Groupings : Segments, Companies, Campaigns.",
+        400
+      );
+  }
+  let contactId = getDestinationExternalID(message, "mauticContactId");
+  let contacts;
+  if (!contactId) {
+    contacts = await searchContactIds(message, Config, endPoint); // Getting the contact Id using Lookup field and then email
+    if (!contacts) {
+      throw new CustomError(
+        "Could not find any contact Id for the given lookup Field or email.",
+        400
+      );
+    }
+    if (contacts.length > 1) {
+      throw new CustomError(
+        "Found more than one Contacts for the given lookupField or email. Retry with unique lookupfield and lookupValue.",
+        400
+      );
+    }
+    if (contacts.length === 1) {
+      const [first] = contacts;
+      contactId = first;
+    }
+  }
+
+  let endpoint = `${endPoint}/${groupClass}/${message.groupId}/contact/${contactId}/add`;
+  if (message.traits.operation === "remove") {
+    endpoint = endpoint.replace("add", "remove");
+  } else if (!message.traits.operation === "add") {
+    throw new CustomError(
+      `${message.traits.operation} is invalid for Operation field. Avaialbale are Add and Remove.`
+    );
+  }
+  const payload = {};
+  return responseBuilder(
+    payload,
+    endpoint,
+    defaultPostRequestConfig.requestMethod,
+    EventType.GROUP,
+    Config
+  );
+};
+
+const identifyResponseBuilder = async (message, destination, endpoint) => {
+  let method;
+  let endPoint;
+  // constructing payload from mapping JSONs
+  const payload = constructPayload(
+    message,
+    mappingConfig[ConfigCategories.IDENTIFY.name]
+  );
+  // if the payload is valid adding address fields if present
+
+  if (validatePayload(payload)) {
+    const { address1, address2 } = deduceAddressFields(message); // throws error if address greater than 128
+    payload.address1 = address1;
+    payload.address2 = address2;
+  }
+  /* 
+     1. if contactId is present  inside externalID we will use that
+     2. Otherwise we will look for the lookup field from the web app 
+  */
+  let contactId = getDestinationExternalID(message, "mauticContactId");
+  let contacts;
+  // searching for contactID from filter options if contactID is not given
+  if (!contactId) {
+    contacts = await searchContactIds(message, destination, endpoint); // Getting the contact Id using Lookup field and then email
```

Comment:
No need for comment here fucntion definition should be enough to explain function behaviuour

Reply:
`contacts = await searchContactIds(message, destination.Config, endpoint)